delfin/drivers/netapp/netapp_fas/netapp_handler.py

```diff
@@ -0,0 +1,605 @@
+            msg = "Failed to ssh netapp_fas %s: %s" % \
+                  (command, six.text_type(e))
+            raise exception.SSHException(msg)
+
+    @staticmethod
+    def change_capacity_to_bytes(unit):
+        unit = unit.upper()
+        if unit == 'TB':
+            res = units.Ti
+        elif unit == 'GB':
+            res = units.Gi
+        elif unit == 'MB':
+            res = units.Mi
+        elif unit == 'KB':
+            res = units.Ki
+        else:
+            res = 1
+        return int(res)
+
+    def parse_string(self, value):
+        capacity = 0
+        if value and value != '' and value != '-':
+            if value.isdigit():
+                capacity = float(value)
+            else:
+                unit = value[-2:]
+                capacity = float(value[:-2]) * int(
+                    self.change_capacity_to_bytes(unit))
+        return capacity
+
+    @staticmethod
+    def handle_detail(system_info, storage_map, split):
+        detail_arr = system_info.split('\r\n')
+        for detail in detail_arr:
+            if detail is not None and detail != '':
+                strinfo = detail.split(split + " ")
+                key = strinfo[0].replace(' ', '')
+                value = ''
+                if len(strinfo) > 1:
+                    value = strinfo[1]
+                storage_map[key] = value
+
+    def get_storage(self):
+        try:
+            STATUS_MAP = {
+                'ok': constants.StorageStatus.NORMAL,
+                'ok-with-suppressed': constants.StorageStatus.NORMAL,
+                'degraded': constants.StorageStatus.ABNORMAL,
+                'unreachable': constants.StorageStatus.ABNORMAL
+            }
+            raw_capacity = total_capacity = used_capacity = free_capacity = 0
+            system_info = self.exec_ssh_command(
+                netapp_constants.CLUSTER_SHOW_COMMAND)
+            version = self.exec_ssh_command(
+                netapp_constants.VERSION_SHOW_COMMAND)
+            status_info = self.exec_ssh_command(
+                netapp_constants.STORAGE_STATUS_COMMAND)
+            version_arr = version.split('\r\n')
+            status = STATUS_MAP.get(status_info.split("\r\n")[2])
+            disk_list = self.get_disks(None)
+            pool_list = self.list_storage_pools(None)
+            storage_map = {}
+            self.handle_detail(system_info, storage_map, split=':')
+            for disk in disk_list:
+                raw_capacity += disk['capacity']
+
+            for pool in pool_list:
+                total_capacity += pool['total_capacity']
+                free_capacity += pool['free_capacity']
+                used_capacity += pool['used_capacity']
+
+            s = {
+                "name": storage_map['ClusterName'],
+                "vendor": netapp_constants.STORAGE_VENDOR,
+                "model": '',
+                "status": status,
+                "serial_number": storage_map['ClusterSerialNumber'],
```

Comment:
Does ClusterSerialNumber unique for storage arrays ?

Reply:
Is the only